apps/server/src/shared/domain/rules/file-element.rule.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { fileElementFactory, roleFactory, setupEntities, userFactory } from '@shared/testing';
+import { Permission } from '../interface';
+import { Actions } from './actions.enum';
+import { FileElementRule } from './file-element.rule';
+
+describe('FileElementRule', () => {
+	let service: FileElementRule;
+	const permissionA = 'a' as Permission;
+	const permissionB = 'b' as Permission;
+
+	beforeAll(async () => {
+		await setupEntities();
+
+		const module: TestingModule = await Test.createTestingModule({
+			providers: [FileElementRule],
+		}).compile();
+
+		service = await module.get(FileElementRule);
+	});
+
+	describe('when user has permissions', () => {
```

Comment:
Negative case when use hasn't got permission is missing

Reply:
It is a dumy rule, without real authorization checks. We will create tests after the concrete implementation.